implementations/rust/ockam/ockam_app/src/invitations/commands.rs

```diff
@@ -118,6 +130,165 @@ pub async fn refresh_invitations<R: Runtime>(app: AppHandle<R>) -> Result<(), St
         let mut writer = invitation_state.write().await;
         *writer = invitations.into();
     }
+    refresh_inlets(&app).await.map_err(|e| e.to_string())?;
     app.trigger_global(REFRESHED_INVITATIONS, None);
     Ok(())
 }
+
+async fn refresh_inlets<R: Runtime>(app: &AppHandle<R>) -> crate::Result<()> {
+    let invitations_state: State<'_, SyncState> = app.state();
+    let reader = invitations_state.read().await;
+    let app_state: State<'_, AppState> = app.state();
+    let cli_state = app_state.state().await;
+    let accepted_invitations_by_node =
+        reader
+            .accepted
+            .iter()
+            .fold(std::collections::HashMap::new(), |mut acc, invitation| {
+                let _ = InletDataFromInvitation::new(&cli_state, invitation).map(|d| {
+                    d.map(|d| {
+                        acc.entry(d.local_node_name.clone())
+                            .or_insert_with(Vec::new)
+                            .push(d);
+                    })
+                });
+                acc
+            });
+    let cli_bin = cli_bin()?;
+    for (node_name, invitations) in accepted_invitations_by_node {
+        let mut node_inlets_to_delete = Vec::new();
+        let node_inlets = {
+            match duct::cmd!(
+                &cli_bin,
+                "tcp-inlet",
+                "list",
+                "--at",
+                &node_name,
+                "--output",
+                "json"
+            )
+            .read()
+            {
+                Ok(json_res) => serde_json::from_str::<Vec<InletStatus>>(&json_res)?,
+                Err(e) => {
+                    warn!(?node_name, %e, "Could not list inlets for node");
+                    // If the node doesn't exist, the command will return an error, so
+                    // we initialize the inlets list to an empty vector.
+                    Vec::new()
+                }
+            }
+        };
+        // Iterate the invitations and create the inlets if they don't exist
+        for invitation_inlet_data in invitations {
+            if let Some(inlet) = node_inlets
+                .iter()
+                // The inlets we create are named after the remote service name (i.e. the remote tcp-outlet name)
+                .find(|inlet| inlet.alias == invitation_inlet_data.service_name)
+            {
+                // If the inlet already exists, mark it to remove it from the node
+                node_inlets_to_delete.push(inlet);
```

Comment:
Does this need one more step or check to ensure that inlets that do already exist and that should still exist afterwards don't get removed by mistake? This seems like it would remove existing ones unconditionally.

Reply:
I've simplified that part. With the naming strategy for the nodes, there is a 1:1 relationship between nodes and tcp-inlets, so the HashMap disapears. I've simplified that part [here](https://github.com/build-trust/ockam/pull/5656/files#diff-fc44c8a5e4b38ca36b52578860084fa796e4931a7b0efdbeb3396c6d36e5ada5R146).

Reply:
That looks clearer to me, thank you.